### PR TITLE
More API validation for quota

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rancher/issues/15583

### CODE REVIEW

File: pkg/api/customization/project/project_store.go [hunk 1]
```diff
@@ -127,12 +129,64 @@ func (s *projectStore) validateResourceQuota(apiContext *types.APIContext, schem
...
 		return err
 	}
-	return httperror.NewFieldAPIError(httperror.MaxLimitExceeded, namespaceQuotaField, fmt.Sprintf("exceeds %s on fields: %s",
-		quotaField, msg))
+
+	if project.ResourceQuota == nil || project.ResourceQuota.UsedLimit == nil {
+		return nil
+	}
+
+	// check that project quota is not bigger than the used quota
```
Comment (kinarashah) [inline pkg/api/customization/project/project_store.go]: I think, project quota is bigger than used quota

File: pkg/api/customization/project/project_store.go [hunk 2]
```diff
@@ -127,12 +129,64 @@ func (s *projectStore) validateResourceQuota(apiContext *types.APIContext, schem
...
+	usedQuotaLimit, err := limitToLimit(project.ResourceQuota.UsedLimit)
+	if err != nil {
+		return err
+	}
+	isFit, msg, err = resourcequota.IsQuotaFit(usedQuotaLimit, []*v3.ResourceQuotaLimit{}, projectQuotaLimit)
```
Comment (kinarashah) [inline pkg/api/customization/project/project_store.go]: maybe, both the isQuotaFit validations common logic into a separate function?

File: pkg/api/customization/project/project_store.go [hunk 3]
```diff
@@ -127,12 +129,64 @@ func (s *projectStore) validateResourceQuota(apiContext *types.APIContext, schem
...
+	if err != nil {
+		return err
+	}
+	if !isFit {
+		return httperror.NewFieldAPIError(httperror.MaxLimitExceeded, namespaceQuotaField, fmt.Sprintf("exceeds used limit on fields: %s",
```
Comment (kinarashah) [inline pkg/api/customization/project/project_store.go]: should be on quotaField here?

Comment (alena1108) [reply pkg/api/customization/project/project_store.go]: you are right

**Pull request merged**